js/web/lib/wasm/jsep/webgpu/ops/gather.ts

```diff
@@ -42,29 +42,29 @@ const createGatherProgramInfo =
       const axisDimLimit = inputShape[axis];
 
       const inputIndices: number[] = [];
-      inputs[1].getBigInt64Array().forEach(v => {
-        let value = Number(v);
+      const inputIndicesArray = indicesElementSize === 2
+        ? inputs[1].getBigInt64Array()
+        : inputs[1].getInt32Array();
+      for (const i of inputIndicesArray[Symbol.iterator]()) {
```

Comment:
could you share what is the benefit of using `array[Symbol.iterator]()` instead of `array`?

Reply:
i had a typescript error with BigInt64Array iteration. but anyway i had to revert these changes because having second input for gather on CPU slows everything down

Reply:
didn't know that "having second input for Gather on CPU slows everything down" part. Is there a model in which Gather takes the second input from a GPU data?

Reply:
I've uploaded it here: https://huggingface.co/aislamov/stable-diffusion-2-1-base-onnx/tree/main
And updated PR where you can load it with JS https://github.com/microsoft/onnxruntime/pull/16749

If you want to test it with JSEP in an easy way, you can checkout dev branch here https://github.com/dakenf/stable-diffusion-webgpu-minimal/tree/dev
Build the runtime and put dist files to node_modules/@aislamov/onnxruntime-web64/dist

Reply:
You'll need to use these build flags: `.\build.bat --config RelWithDebInfo --build_wasm --skip_tests --compile_no_warning_as_error --enable_wasm_simd --use_jsep --parallel --enable_wasm_memory64` threads with 64bit memory are not yet rolled out to chrome canary

Reply:
I've tested it only once and performance fell from ~2.2 sec to 3+sec
However, as i don't trust computers at all, so that could be anything from IDE indexing files to misaligned stars positions in the sky. Can do some more tests tomorrow

Reply:
From a quick check, I noticed that most `Gather` nodes are used in this pattern:

![image](https://github.com/microsoft/onnxruntime/assets/7679871/8ac7d30d-2742-4570-a591-bcf5033b1e7a)

In this pattern, all nodes that included in the red circle are expected to run on CPU, not GPU. Because `Shape` generates its output (shape of the tensor) putting on CPU, and followed by a few very light-weighted operators dealing with int64 types (which are mostly used as processing indices/dimensions/shapes, because we don't use int64 for weights usually), and eventually being taken as input[1] for a `Reshape` node, which is also on CPU.

Reply:
I am making this assumption: most (if not all) int64 types should be put on CPU for calculating. I will try to figure out if this assumption is correct. But for this example, if we unregister Gather.int64/Mul.int64/Div.int64/Unsqueeze.int64/Concat.int64 for webgpu, I expect the whole model get better performance overall.

Reply:
By the way, this is a separated discussion that does not block the PR.

Reply:
Yes. This PR does not allow any Gather (and some other) nodes with int64 input on GPU. My reverted changes affect only input[0] with float32, so most likely when first input is float32, second input is from some other node processed by GPU. Will double-check that tomorrow